utils/aws-operations.js

```diff
@@ -0,0 +1,93 @@
+const AWS = require('aws-sdk');
+const config = require('config');
+
+const {
+  accessKeyId,
+  secretAccessKey,
+  region,
+  endpoint,
+  s3ForcePathStyle,
+} = config.get('aws');
+
+const s3 = new AWS.S3({
+  apiVersion: '2006-03-01',
+  accessKeyId,
+  secretAccessKey,
+  region,
+  endpoint,
+  s3ForcePathStyle,
+});
+this.bucket = null;
```

Comment:
I could be wrong, but by using `this` alone, looks like it'll refer to the global object right? Is that's what you wanna do, shouldn't we store this variable to maybe `app.local`?

Reply:
Since `this` is being used inside a module, I don't think it's referring to the global object. I tried `console.log(this.bucket)` in `aws-operations.js`, `validate-data-source.js` (after calling `setBucket()`), and in `app.js` (after `validateDataSource()` finished resolving). The result was the correct value in `aws-operations.js` and `undefined` in the other two. I found this https://stackoverflow.com/a/22771432 after looking into it some more and it looks like using `this` in a module is synonymous with `module.exports`. The usage of `this.bucket` was just meant for a value that is only going to be used by `aws-operations.js`. Looking at it more, I think I could just make this a variable in the global scope of `aws-operations.js`. What are your thoughts on this?